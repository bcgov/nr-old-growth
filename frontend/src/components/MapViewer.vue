<template>
  <div
    id="mapContainer"
    style="height: 90vh; width: 101%; position: relative"
  />
</template>

<script lang="ts">
import { defineComponent } from "vue";
import * as L from "leaflet";
import { EsriProvider, GeoSearchControl } from "leaflet-geosearch";
import "@geoman-io/leaflet-geoman-free";
import "leaflet/dist/leaflet.css";
import "@geoman-io/leaflet-geoman-free/dist/leaflet-geoman.css";
import "leaflet-geosearch/dist/geosearch.css";

export default defineComponent({
  data() {
    return {
      zoom: 12,
      map: null as any,
    };
  },
  mounted() {
    const container = document.getElementById("mapContainer");
    if (container) {
      // initial map view
      this.map = L.map("mapContainer").setView([54.7276, -127.6476], 12);
      new L.TileLayer(
        "https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}"
      ).addTo(this.map);

      // add search control
      const provider = new EsriProvider();
      const searchControl = new (GeoSearchControl as any)({
        provider,
        // style: 'bar'
      });
      this.map.addControl(searchControl);

      // L.marker([54.7276, -127.6476]).addTo(this.map);

      const priorityLayers = L.tileLayer
        .wms("https://openmaps.gov.bc.ca/geo/ows", {
          format: "image/png",
          transparent: true,
          layers: "WHSE_FOREST_VEGETATION.OGSR_TAP_PRIORITY_DEF_AREA_SP",
        })
        .addTo(this.map);

      // add layer control
      const overlays = {
        PRIORITY_DEF_AREA: priorityLayers,
      };
      L.control.layers({}, overlays).addTo(this.map);

      // add drawing control
      this.map.pm.addControls({
        position: "topright",
        // drawCircle: false,
      });
    }
  },
  methods: {},
});
</script>

<style>
/* ------------- Draw control ------------- */
.leaflet-pm-toolbar.leaflet-pm-draw {
  position: fixed;
  top: 130px;
  right: 3px;
}

.leaflet-pm-toolbar.leaflet-pm-edit {
  position: fixed;
  top: 330px;
  right: 3px;
}
</style>
