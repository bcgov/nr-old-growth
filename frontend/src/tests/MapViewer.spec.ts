import { nextTick } from "vue";
import { describe, it, expect } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";

import MapViewer from "../components/MapViewer.vue";

describe("MapViewer", () => {
  it("renders properly", async () => {
    const wrapper = mount(MapViewer);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    // expect(wrapper.find("#mapContainer").exists()).toBe(true);
  });
});
