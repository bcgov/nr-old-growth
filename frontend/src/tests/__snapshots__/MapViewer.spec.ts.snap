// Vitest Snapshot v1

exports[`MapViewer > renders properly 1`] = `"<div id=\\"mapContainer\\" style=\\"height: 90vh; width: 101%; position: relative;\\"></div>"`;
